Docs update (#515)
fiberplane/fiberplane: www/src/content/docs/docs/features/playground.mdx

---
title: Playground Features
description: Explore and test your API with Fiberplane Playground
order: 3
---

import { Aside, Steps, Tabs, TabItem } from "@astrojs/starlight/components";

Fiberplane Playground provides an interactive interface for exploring and testing your Hono API. This guide covers the main features of the Playground UI and how to use them effectively.

## API Documentation Browser

The Playground generates interactive documentation from your OpenAPI specification. You can:

> - Browse all available endpoints grouped by paths

Reviewer: ```suggestion
- Browse all available endpoints, grouped by their tags
```

- View detailed request and response schemas

> - See example values based on your schema definitions

Reviewer: ```suggestion
- Test example payloads based on your schema definitions
```

- Explore authentication requirements for each endpoint

<figure class="light:sl-hidden">

>   ![Playfround Documentation

Reviewer: ```suggestion
  ![Playground Documentation
```

  Browser](@/assets/docs/playground-documentation-browser-dark.png)
</figure>

<figure class="dark:sl-hidden">
  ![Playground Documentation
  Browser](@/assets/docs/playground-documentation-browser.png)
</figure>

## Making API Requests

The Playground provides a user-friendly interface for constructing API requests:

1. **Select an Endpoint**: Choose the endpoint you want to test from the sidebar
2. **Configure Request Parameters**:
   - Path parameters
   - Query parameters
   - Request body (automatically populated with example data from your schema)

### Example Data

The Playground automatically generates example data based on your OpenAPI schemas. For instance, using the user creation example:

```json
{
  "name": "Matthew",
  "email": "matthew@cloudflare.com"
}
```

You can:

- Use the provided example data as is
- Modify the data to test different scenarios

> - View the expected response format

Reviewer: ```suggestion
- Change the content-type of the request
```

### Response Viewer

After making a request, you can:

- View the response status code
- Inspect response headers
- See the formatted response body

> - View response timing information

Reviewer: ```suggestion
```

<figure class="light:sl-hidden">
  ![Playground: Send Requests](@/assets/docs/playground-send-request-dark.png)
</figure>

<figure class="dark:sl-hidden">
  ![Playground: Send Requests](@/assets/docs/playground-send-request.png)
</figure>

## Authentication

> The Playground supports authentication:

Reviewer: ```suggestion
The Playground supports reusable Bearer authentication:
```
